packages/internals/support/src/emitting-promise.js

```diff
@@ -0,0 +1,30 @@
+const {EventEmitter2} = require('eventemitter2');
+const assert = require('check-types').assert;
+
+class EmittingPromise extends Promise {
+  constructor(resolver) {
+    assert.function(resolver, `EmittingPromise constructor requires a resolver with params 'resolve, reject [, emit]' [invalid-resolver]`);
+    const emitter = new EventEmitter2();
```

Comment:
If we want to support wildcards you need to explicitly enable it in EE2 via `{wildcard: true}` ... not sure why it's not the default behaviour tbh! :-)